_episodes/l3_03_test_fixture.md

```diff
@@ -0,0 +1,439 @@
+
+Let us understand how to create a test fixture, setup and teardown functions with examples.
+
+## 2. Problem under consideration for testing
+
+In order to create our own test fixture, we will first explain the context or problem that we are trying to solve. We will be adding tests to check that our code works as intended by making use of test fixtures.
+
+Consider that you want to write a program to manage the details of an employee. The program should allow you to add basic details of an employee such as:
+
+1. Name
+2. Age
+3. Basic Salary
+4. Number of years of employment
+5. Basic Bonus the employee has received in this year.
+
+The program should calculate the Net Bonus, Tax and Salary based on the following rules:
+
+1. Bonus Rule: An employee gets an additional £1000 bonus if she or he has worked for more than 10 years.
+2. Tax Rule: Tax is calculated on a combination of basic salary and net bonus as shown below.
+
+    * `0` if salary is less than `10k` GBP.
+    * `10%` for salary between `10K-20K` GBP.
+    * `20%` for salary between `20K-50K` GBP
+    * `50%` for salary greater than `50K` GBP
+
+Based on above, we can declare the employee class in `employee.h` as shown below ([Declaration of Employee class](../code/Chapter3/employee.h)).
+
+```cpp
+class Employee 
+{
+private:
+    string name;
+    float age;
+    double base_salary; //salary before calculating tax and adjusting bonus.
+    double number_years_employed;
+
+    double basic_bonus; //bonus for current year.
+    double net_bonus; //bonus after adjusting for experience.
+    
+    double tax_amount;
+    double net_salary; //salary after calculating tax and adjusting bonus.
+
+public:
+    // Constructor.
+    Employee(const string& employee_name, float employee_age, 
+             double employeeSalary, double employeeNumberYearsEmployed,
+             double employeeBonus);
+
+    // Public member functions to set values.
+    void setName(const std::string& employee_name); 
+    void setAge(float employee_age) ;
+    void setBaseSalary(double employeeSalary); 
+    void setNumberYearsEmployed(double employeeNumberYearsEmployed); 
+    void SetBasicBonus(double employeeBonus);
+
+    void calcNetBonus(); //To calculate net bonus while considering experience.
+    void calcTaxAmount(); // To calculate tax to be paid based on salary with bonus
+    void calcNetSalary(); // To calculate net salary after adjusting tax and bonus.
+
+    // Getter functions.
+    string getName() const;
+    float getAge() const;
+    double getBasicSalary() const;
+    double getNumberYearsEmployed() const;
+    double getBasicBonus() const;
+    double getNetBonus() const;
+    double getTaxAmount() const;
+    double getNetSalary() const;
+
+    void displayInfo() const; 
+
+    // Destructor
+    ~Employee();
+};
+```
```

Comment:
I think this block of code can be replaced by the following, keeping code and lesson in sync. See https://github.com/stevemar/code-reference-in-readme/tree/main

I'm not sure how that would be rendered by GitHub Pages, though. Maybe we can give it a try and, if it does not work, just put the code back again.

```suggestion
https://github.com/ImperialCollegeLondon/lkr_unit_testing_forked/blob/df3743fc160148b0b6b2bd59cca4cd511b766008/code/Chapter3/employee.h#L11-L56
```

Reply:
I tried to accept your suggestion in 857ffa697ca3e43fdfec33db7c7d42f4e6c8eb91 but then I reverted it back in 84374c9d6921110a75a42537221ce7898e57c24f. This is because only a few lines were being shown in the box and there has to be a lot of scrolling in the box to explain the code.